scripts/core/editor3/components/Editor3Component.tsx

```diff
@@ -197,10 +197,6 @@ export class Editor3Component extends React.Component<IProps, IState> {
     }
 
     spellcheck() {
```

Comment:
Why are you changing this? Isn't it better to have the condition in one place vs multiple?

Reply:
I don't use this spellchecker anymore but I think it's a good practice to do this as it can prevent other conditions that affect performance. For example, [this condition](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/editor3/components/Editor3Component.tsx#L495) can be heavy on performance and it might not even be necessary if spellchecker is disabled.

Reply:
I get what you're trying to do now. You could keep the condition in one place and have the same performance if you passed the expensive function as a callback which would only get executed if the spellchecker is enabled.

```
spellcheck(() => prevProps.editorState.getCurrentContent() !== this.props.editorState.getCurrentContent());
```

On the other hand, draft-js is using immutable-js, so I expect that when you call `getCurrentContent` it simply returns an reference to the object and doesn't do any processing, no?